core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -866,13 +877,19 @@ public SchemaString get(int id, boolean fetchMaxId) throws SchemaRegistryExcepti
     }
     SchemaString schemaString = new SchemaString();
     schemaString.setSchemaType(schema.getSchemaType());
-    schemaString.setSchemaString(schema.getSchema());
-    List<SchemaReference> refs = schema.getReferences();
-    if (refs != null) {
-      schemaString.setReferences(refs.stream()
-          .map(ref -> new io.confluent.kafka.schemaregistry.client.rest.entities.SchemaReference(
-              ref.getName(), ref.getSubject(), ref.getVersion()))
-          .collect(Collectors.toList()));
+    List<io.confluent.kafka.schemaregistry.client.rest.entities.SchemaReference> refs =
+        schema.getReferences() != null
+        ? schema.getReferences().stream()
+            .map(ref -> new io.confluent.kafka.schemaregistry.client.rest.entities.SchemaReference(
+                ref.getName(), ref.getSubject(), ref.getVersion()))
+          .collect(Collectors.toList())
+        : null;
+    schemaString.setReferences(refs);
+    if (format != null && !format.trim().isEmpty()) {
+      ParsedSchema parsedSchema = parseSchema(schema.getSchemaType(), schema.getSchema(), refs);
+      schemaString.setSchemaString(parsedSchema.formattedString(format));
+    } else {
```

Comment:
I think it might be a bit nicer to have another overload for the default format: 'text' (in addition to empty or null).

In the clients, I feel like I want an enum for specifying this: `SchemaFormat.Text/Serialized`. Less flexible structure than a string, but I don't feel it'll ever be too limiting?

Reply:
I was also thinking of possibly having a PRETTY_PRINT option as well, in the future.